lls/cache.c

```diff
@@ -360,6 +433,40 @@ lls_process_reqs(struct lls_config *lls_conf)
 			}
 			break;
 		}
+		case LLS_REQ_PING6: {
+			struct lls_ping6_req *ping6 = &reqs[i]->u.ping6;
+			uint16_t tx_queue;
+			struct token_bucket_ratelimit_state *rs;
+			int num_granted_pkts;
+			int i;
+			int num_reply = 0;
+
+			if (ping6->iface == &lls_conf->net->front) {
+				tx_queue = lls_conf->tx_queue_front;
+				rs = &lls_conf->front_icmp_rs;
+			} else {
+				tx_queue = lls_conf->tx_queue_back;
+				rs = &lls_conf->back_icmp_rs;
+			}
+
+			num_granted_pkts =
+				tb_ratelimit_allow_n(ping6->num_pkts, rs);
+
+			for (i = 0; i < ping6->num_pkts; i++) {
+				int ret = xmit_icmpv6_reply(
+					ping6->iface, ping6->pkts[i]);
+				if (ret == 0 && num_reply < num_granted_pkts) {
```

Comment:
This `if` statement is not well structured because `xmit_icmpv6_reply()` is called even when `num_reply >= num_granted_pkts`, which implies that the packet is going to be dropped anyway. My take on this is to have the body of the `for` loop as follows:
```
if (num_reply >= num_granted_pkts || xmit_icmpv6_reply(ping6->iface, ping6->pkts[i]) != 0) {
	rte_pktmbuf_free(ping6->pkts[i]);
	continue;
}
ping6->pkts[num_reply++] = ping6->pkts[i];
```

See analogous case in `case LLS_REQ_PING:` in the next patch.